packages/hash/api/src/graphql/resolvers/taskExecutor/index.ts

```diff
@@ -22,3 +32,217 @@ export const executeDemoTask: Resolver<
     }
   }
 };
+
+export const executeGithubSpecTask: Resolver<
+  Promise<string>,
+  {},
+  GraphQLContext
+> = async (_, __, { dataSources: { taskExecutor } }) => {
+  if (!taskExecutor) {
+    throw new ApolloError(
+      "A task-executor wasn't started, so external tasks can't be started",
+    );
+  } else {
+    try {
+      return await taskExecutor
+        .run_task(Tasks.GithubSpec)
+        .then((res) => JSON.stringify(res));
+    } catch (err: any) {
+      throw new ApolloError(`Task-execution failed: ${err}`);
+    }
+  }
+};
+
+export const executeGithubCheckTask: Resolver<
+  Promise<string>,
+  {},
+  GraphQLContext,
+  MutationExecuteGithubCheckTaskArgs
+> = async (_, { config }, { dataSources: { taskExecutor } }) => {
+  if (!taskExecutor) {
+    throw new ApolloError(
+      "A task-executor wasn't started, so external tasks can't be started",
+    );
+  } else {
+    try {
+      return await taskExecutor
+        .run_task(Tasks.GithubCheck, config)
+        .then((res) => JSON.stringify(res));
+    } catch (err: any) {
+      throw new ApolloError(`Task-execution failed: ${err}`);
+    }
+  }
+};
+
+/** @todo - Make Airbyte types available in api package */
+type AirbyteCatalog = Array<{
+  name?: string;
+  json_schema?: JSONObject;
+}>;
+
+type AirbyteRecords = Array<{
+  /**
+   * the name of this record's stream
+   */
+  stream: string;
+  /**
+   * the record data
+   */
+  data: {
+    [k: string]: unknown;
+  };
+  /**
+   * when the data was emitted from the source. epoch in millisecond.
+   */
+  emitted_at: number;
+  /**
+   * the namespace of this record's stream
+   */
+  namespace?: string;
+  [k: string]: unknown;
+}>;
+
+const ExistingEntityChecker = async (db: DbAdapter, user: User) => {
+  const streamsWithEntityTypes: Map<string, string> = new Map();
+  const dbTypes = await db.getAccountEntityTypes({
+    accountId: user.accountId,
+  });
+
+  return {
+    getExisting: (entityTypeName: string) => {
+      // check the map to find the entityTypeId associated with the stream if we've already encountered it
+      let entityTypeId: string | undefined | null =
+        streamsWithEntityTypes.get(entityTypeName);
+
+      if (entityTypeId === undefined) {
+        // check all entityTypes to see if there's one with the same name as the stream
+        entityTypeId = dbTypes.find(
+          (entityType) => entityType.metadata.name === entityTypeName,
+        )?.entityId; /** @todo - WHY is this entityId instead of entityTypeId?! */
```

Comment:
Right but I don't understand why the entityTypeId _on an EntityType_ is on the `entityId` field? 

Is it because `entityId` is an alias for `id` and I'm therefore actually calling `entityType.id` (which makes sense) and just being mislead?

Reply:
I think we moved away from `id` as it lead to confusion, and we want to specify what id we're referring to from any type. 
This instance of `id` could just have been an oversight.

If I recall correctly, `entityTypeId` actually refers to 'the' `EntityType`  system type when resolved in the GQL API. That makes it so that the fields on the `EntityType` model class aren't called `entityType(Version)Id`.
This system type also referred to as the `EntityTypeType` in some places. I find this generally confusing.

Reply:
I agree with all of that, I'm just still uncertain as to what we're actually discussing as an action item here. The state of things right now is that on an `EntityType`(e.g. the objects returned by `getAccountEntityTypes`) you have:
* `entityType.entityTypeId := undefined`
* `entityType.entityId := [the entityType ID]`

This clearly isn't right, I don't even know why the `EntityType` type has an `entityId` or `entityVersionId` property as it's not an entity.

Reply:
This question was more of an observation when first writing.
As an action item, I would like for us to capture these issues in an asana task (there might be one already?), as they are overlapping, and since it does lead to some confusing debugging sessions while writing code around EntityTypes.

Reply:
Captured in (asana task)[https://app.asana.com/0/1200211978612931/1200809642643269/f] (_internal_)

Reply:
Cheers, I've updated the todo to include a link to it

Reply:
Good point, Ben. I think the comment on the field refers to is not the field that is actually set though.
It says `alias of 'entityId'` but it is set as `id: this.entityVersionId,` in the model class.